code/game/objects/items/tools/multifunctiontool.dm

```diff
@@ -1,76 +1,89 @@
-//Power Tools
-/obj/item/powertool
-	var/stateswap = 0 //what state is it in
-	var/icon_state_swap = 0 //which icon should it be
-	var/tool_behaviour_swap = 0 //what tool behaviour(s) should be here
-	var/swapsound = 0 //the sound that plays for swapping
+//Multifunction tools
+/obj/item/multifunctiontool
+	var/list/tool_icon_states = list() //which icon should it be
+	var/list/tool_behaviours = list()  //what tool behaviour(s) should be here
+	var/list/tool_mode_names = list() //text added to the examine description
+	var/active_behaviour_index = 1
+	var/swapsound //the sound that plays for swapping
 
-/obj/item/powertool/attack_self(mob/living/carbon/user) //activating the item causes it to swap icons and behaviours
-	stateswap = !stateswap
-	if(stateswap)
-		icon_state = icon_state_swap
-		tool_behaviour = tool_behaviour_swap
-	else
-		icon_state = initial(icon_state)
-		tool_behaviour = initial(tool_behaviour)
+/obj/item/multifunctiontool/attack_self(mob/living/carbon/user) //activating the item causes it to swap icons and behaviours
+	if(tool_behaviours.len > 1)
+		select_mode(user)
+	tool_behaviour = tool_behaviours[active_behaviour_index]
+	icon_state = tool_icon_states[active_behaviour_index]
 	playsound(get_turf(user), swapsound, 50, 1)
-	to_chat(user, "<span class='notice'>You attach a different bit to [src].</span>")
+	to_chat(user, "<span class='notice'>You use a different part on [src].</span>")
+
+/obj/item/multifunctiontool/proc/select_mode(mob/living/user)
+	active_behaviour_index++
+	if(active_behaviour_index > tool_behaviours.len)
+		active_behaviour_index = 1
 	return
+
+/obj/item/multifunctiontool/examine(mob/user)
+	. = ..()
+	. += "[tool_mode_names[active_behaviour_index]]"
```

Comment:
Does this properly add spaces/punctuation? I.e. does this not give you `Power DrillScrewdriver bit attached`?